index.js

```diff
@@ -260,10 +244,26 @@ morgan.token('date', function getDateToken (req, res, format) {
  * response status code
  */
 
-morgan.token('status', function getStatusToken (req, res) {
-  return headersSent(res)
-    ? String(res.statusCode)
-    : undefined
+morgan.token('status', getStatusToken)
+
+/**
+ * colored response status code
+ */
+
+morgan.token('status-colored', function getStatusColoredToken (req, res) {
+  var status = getStatusToken(req, res)
+
+  // get status color
+  var color = status >= 500 ? 31 // red
+    : status >= 400 ? 33 // yellow
+      : status >= 300 ? 36 // cyan
+        : status >= 200 ? 32 // green
+          : 0 // no color
+
+  // using placeholder here since `compile` adds it if token function returns falsy,
+  // but this coloring token returns color data anyway (as per tests) and
+  // value won't be eligible for substitute
```

Comment:
I'm not sure why we should do this vs returning undefined and allow the compile function to take care of it. Does the actual color of the dev format change at all if we do this? If an unknown status returns something here, then this would be an outlier for how tokens currently work and likely needs explicit docs around this. But I don't actually think it's necessary, but happy to be proven wrong there :)

Reply:
this is a great point and simplified the code :)